src/syntax-tree/module.ts

```diff
@@ -573,4 +576,58 @@ export class Module {
 
         return ret ?? CodeStatus.Runnable;
     }
+
+    performActionOnBFS(duringAction: (code: CodeConstruct) => void) {
```

Comment:
Variable type calculations and a few other places should be refactored to use this, if possible, instead of constantly reimplementing the same BFS everywhere.